### PR TITLE
Fix continuous ttest

### PR DESCRIPTION
**Pull request by regislebrun**
This PR fixes the p-value computation in ContinuousTTest and improves its performance by working with the log-PDF instead of the PDF of the Bernstein copulas.
It also provides a slight improvement of MixedHistogramUserDefined by using OT::DistFunc::rDiscrete() instead of a custom implementation of the alias method.

### CODE REVIEW

**Comment by jschueller** [comment]
the test t_papillon seems to hang (>10m), I just reran Travis just to be sure

File: lib/src/ContinuousTTest.cxx [hunk 1]
```diff
@@ -119,106 +128,112 @@ double ContinuousTTest::getTTest(const OT::UnsignedInteger Y,
...
+  else
+    {
+      for (unsigned int i = 0; i < N; ++i)
+	{
+	  logDenominator = logFYZX[i] + logFX[i];
+	  if (logDenominator > smallLog)
+	    {
+	      H += std::pow(-std::expm1(0.5 * (logFYX[i] + logFZX[i] - logDenominator)), 2.0);
+	      double gX = 1.0;
+	      for (int j = 0; j < d; ++j)
```
Comment (jschueller) [inline lib/src/ContinuousTTest.cxx]: j is unsigned

Comment (regislebrun) [reply lib/src/ContinuousTTest.cxx]: @jschueller fixed

**Comment by regislebrun** [comment]
@jschueller @phwuil The papillon test is painfully slow. Nearly 8mn on my laptop... I am checking with perf/flamegraph what is going on.

**Comment by jschueller** [comment]
As you already noticed, most of the time is spent in EmpiricalBernsteinCopula::computelogpdf, it will be probably worth rewriting the for loop by wrapping fmath::expd_v too in SpecFunc.
But it's difficult to see why it fails with this PR.

**Comment by regislebrun** [comment]
@jschueller The reason why the test was taking so much time compared with the previous version is the correction of the bug in the computation of the p-value in ComputeTTest, nothing to do with rDiscrete. BTW, the use of rDiscrete gives a small boost to the sampling method of the JunctionTreeBernsteinCopula class.

**Comment by jschueller** [comment]
your file t_papillon.expout must be removed

**Pull request merged**